src/backend/marsha/core/xapi.py

```diff
@@ -171,45 +191,54 @@ def from_lti(self, document, statement, jwt_token):
 class XAPIVideoStatement(XAPIStatementMixin):
     """Object managing statement for video objects."""
 
-    def _get_activity_type(self, video):
-        """Return the activity type for a given video"""
-
-        activity_type = "https://w3id.org/xapi/video/activity-type/video"
-
+    def _get_object(self, video):
+        """Return the object xAPI instance for a given video"""
         # When the video is a live we change the activity to webinar
         if video.is_live:
-            activity_type = "http://id.tincanapi.com/activitytype/webinar"
-
-        return activity_type
+            return WebinarActivity(
+                id=f"uuid://{video.id}",
+                definition=WebinarActivityDefinition(
+                    name={self.get_locale(): video.title}
+                ),
+            ).model_dump(exclude_none=True)
+
+        return VideoActivity(
+            id=f"uuid://{video.id}",
+            definition=VideoActivityDefinition(name={self.get_locale(): video.title}),
+        ).model_dump(exclude_none=True)
 
     # pylint: disable=too-many-arguments
-    def _build_statement(self, video, statement, homepage, user=None, user_id=None):
+    def _build_statement(
+        self, video, statement, homepage, user=None, user_id=None, username=None
+    ):
         """Build all common properties for a video."""
         if re.match(r"^http(s?):\/\/.*", homepage) is None:
             homepage = f"http://{homepage}"
 
         statement = self.build_common_statement_properties(
-            statement, homepage, user=user, user_id=user_id
-        )
-
-        category_id = (
-            "https://w3id.org/xapi/lms"
-            if statement["verb"]["id"] == "http://id.tincanapi.com/verb/downloaded"
-            else "https://w3id.org/xapi/video"
+            statement, homepage, user=user, user_id=user_id, username=username
         )
 
-        statement["context"].update(
-            {"contextActivities": {"category": [{"id": category_id}]}}
-        )
+        if statement["verb"]["id"] == "http://id.tincanapi.com/verb/downloaded":
+            statement["context"].update(
+                {
+                    "contextActivities": {
+                        "category": [LMSProfileActivity().model_dump(exclude_none=True)]
+                    }
+                }
+            )
+        else:
+            statement["context"].update(
+                {
+                    "contextActivities": {
+                        "category": [
+                            VideoProfileActivity().model_dump(exclude_none=True)
+                        ]
+                    }
+                }
+            )
```

Comment:
Same here, would it be possible to work with ralph pydantic models instead of modifying a dict directly?
We could use them without any validation using a `.construct()`